tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -418,51 +418,131 @@ static PyObject * THPVariable_cuda(PyObject* self, PyObject* args, PyObject* kwa
   END_HANDLE_TH_ERRORS
 }
 
-static PyObject * THPVariable_to_type(PyObject* self, ScalarType scalarType) {
+static PyObject * THPVariable_to_type(PyObject* self, ScalarType scalarType, c10::optional<c10::MemoryFormat> optional_memory_format) {
   HANDLE_TH_ERRORS
   auto& self_ = reinterpret_cast<THPVariable*>(self)->cdata;
-  // Setting to MemoryFormat::Contiguous now, will change to accept memory_format in next PR
-  return THPVariable_Wrap(dispatch_to(self_, scalarType, false, false, MemoryFormat::Contiguous));
+  return THPVariable_Wrap(dispatch_to(self_, scalarType, false, false, optional_memory_format));
   END_HANDLE_TH_ERRORS
 }
-static PyObject * THPVariable_byte(PyObject* self, PyObject* args) {
```

Comment:
I am a bit skeptical that these should be manually bound. They seem like fairly simple functions that the code generator should be able to understand...

Reply:
Eh, if we made `byte`, `char`, ..., into native functions, then they'd call `at::to`. This would do two dispatches; once for `byte` and once for `to`, which could be annoying. I don't know how much times dispatches take nowadays.

Reply:
If we care about the performance, we can call `native::to` and specify the derivative for each of `byte`, `char`, ...

Reply:
I don't think the dispatching really matters, casting isn't super common and these functions don't even support non_blocking, so you are going to pay the copy immediately anyway.